sim/aviron/src/main.zig

```diff
@@ -21,41 +21,51 @@ pub fn main() !u8 {
         return if (cli.options.help) @as(u8, 0) else 1;
     }
 
-    // Emulate Atmega382p device size:
-
     // TODO: Add support for more MCUs!
     std.debug.assert(cli.options.mcu == .atmega328p);
+    const mcu_config = aviron.mcu.atmega328p;
 
-    var flash_storage = aviron.Flash.Static(32768){};
-    var sram = aviron.RAM.Static(2048){};
-    var eeprom = aviron.EEPROM.Static(1024){};
+    var flash_storage = aviron.Flash.Static(mcu_config.flash_size){};
+    var sram = aviron.RAM.Static(mcu_config.sram_size){};
+    var eeprom = aviron.EEPROM.Static(mcu_config.eeprom_size){};
     var io = IO{
         .sreg = undefined,
-        .sp = sram.data.len - 1,
+        .sp = mcu_config.sram_base + mcu_config.sram_size - 1,
     };
 
+    // Create memory interfaces
+    const flash_mem = flash_storage.memory();
+
+    // Build Device views
+    const io_dev = io.device();
+    const sram_dev = sram.device();
+    const eeprom_dev = eeprom.device();
+
+    var spaces = try aviron.mcu.build_spaces(allocator, mcu_config, sram_dev, io_dev, eeprom_dev);
+    defer spaces.deinit(allocator);
+
     var cpu = aviron.Cpu{
         .trace = cli.options.trace,
 
-        .flash = flash_storage.memory(),
-        .sram = sram.memory(),
-        .eeprom = eeprom.memory(),
-        .io = io.memory(),
+        .flash = flash_mem,
+        .data = spaces.data.device(),
+        .io = spaces.io.device(),
+        .eeprom = spaces.eeprom.device(),
 
-        .code_model = .code16,
-        .instruction_set = .avr5,
+        .code_model = mcu_config.code_model,
+        .instruction_set = mcu_config.instruction_set,
 
         .sio = .{
-            .ramp_x = null,
-            .ramp_y = null,
-            .ramp_z = null,
-            .ramp_d = null,
-            .e_ind = null,
+            .ramp_x = mcu_config.special_io.ramp_x,
+            .ramp_y = mcu_config.special_io.ramp_y,
+            .ramp_z = mcu_config.special_io.ramp_z,
+            .ramp_d = mcu_config.special_io.ramp_d,
+            .e_ind = mcu_config.special_io.e_ind,
 
-            .sp_l = @intFromEnum(IO.Register.sp_l),
-            .sp_h = @intFromEnum(IO.Register.sp_h),
+            .sp_l = mcu_config.special_io.sp_l,
+            .sp_h = mcu_config.special_io.sp_h,
 
-            .sreg = @intFromEnum(IO.Register.sreg),
+            .sreg = mcu_config.special_io.sreg,
         },
```

Comment:
this should be `.sio = mcu_config.special_io` and having both parts use the same type

Reply:
Done